benchmark/text-generation/benchmark.py

```diff
@@ -9,57 +9,46 @@
 from optimum.neuron import NeuronModelForCausalLM
 
 
-def generate(model, input_ids, length):
+def generate(model, input_ids, max_new_tokens):
     start = time.time()
     with torch.inference_mode():
-        output_tokens = model.generate(input_ids, do_sample=False, min_length=length, max_length=length)
+        output_tokens = model.generate(input_ids, do_sample=False, max_new_tokens=max_new_tokens)
     end = time.time()
     return output_tokens, (end - start)
 
 
 def run(model_id, inc_length, max_length, json_path=None):
-    prompts = ["One of my fondest memory"]
+    # Encode the reference prompt
+    with open("./wiki.txt") as f:
+        prompt = f.read()
+    tokenizer = AutoTokenizer.from_pretrained(model_id)
+    tokens = tokenizer([prompt], return_tensors="pt")
+    # Evaluate the batch size
     config = AutoConfig.from_pretrained(model_id)
     batch_size = config.neuron["batch_size"]
-    if len(prompts) < batch_size:
-        prompts = prompts + [prompts[-1]] * (batch_size - len(prompts))
     model = NeuronModelForCausalLM.from_pretrained(model_id, export=False, low_cpu_mem_usage=True)
-    tokenizer = AutoTokenizer.from_pretrained(model_id)
-    # Specify padding options for decoder-only architecture
-    tokenizer.pad_token_id = tokenizer.eos_token_id
-    tokenizer.padding_side = "left"
-    # Encode the first input tokens
-    tokens = tokenizer(prompts, return_tensors="pt", padding=True)
-    bootstrap_input_ids = tokens.input_ids
-    # Generate the first set of inputs
-    input_ids, latency = generate(model, bootstrap_input_ids, inc_length)
-    input_length = input_ids.size()[-1]
+
+    def get_input_ids(tokens, batch_size, input_length):
+        return tokens.input_ids[:, :input_length].repeat((batch_size, 1))
```

Comment:
nit: if `tokens.input_ids` initially has a batch size > 1 the output will have a batch size of `batch_size x original_batch_size`, maybe just select the first batch before the repeat.